### PR TITLE
Pass kernel size and kernel type and padding as init arguments

### PR DESCRIPTION
**Pull request by st9007a**
1. Add `kernel_size`, `kernel_type`, and `padding` as init arguments of KIN.
   - kernel_size: The size of kernel used to kernelize.
   - kernel_type: The type of kernel. 'constant' or 'gaussian'.
   - padding: The padding size of cache table.
2. Move `KIN_PADDING` and `KIN_KERNEL` into `NORMALIZATION` and rename to `PADDING` and `KERNEL_TYPE`.
3. Fix norm_cfg error.

### CODE REVIEW

File: models/kin.py [hunk 1]
```diff
@@ -5,10 +5,36 @@
...
+        kernel_size=3,
+        kernel_type='gaussian',
+        padding=1,
+        eps=0,
+        affine=True,
+        device="cuda",
+    ):
         super(KernelizedInstanceNorm, self).__init__()
 
+        if kernel_size % 2 == 0:
```
Comment (Kaminyou) [inline models/kin.py:33]: Additionally check `padding  * 2 + 1 == kernel_size`?

File: F-LSeSim/models/kin.py [hunk 2]
```diff
@@ -5,10 +5,36 @@
...
+        kernel_size=3,
+        kernel_type='gaussian',
+        padding=1,
+        eps=0,
+        affine=True,
+        device="cuda",
+    ):
         super(KernelizedInstanceNorm, self).__init__()
 
+        if kernel_size % 2 == 0:
```
Comment (Kaminyou) [inline F-LSeSim/models/kin.py:33]: Additionally check `padding * 2 + 1 == kernel_size`?

**Review by Kaminyou — COMMENTED** [review]
Additionally check `padding * 2 + 1 == kernel_size`?

File: models/kin.py (hunk 1 above)
Comment (st9007a) [reply models/kin.py:33]: As discussion, I will remove padding from init arguments. The padding value will be derived from kernel_size.

Comment (st9007a) [reply models/kin.py:33]: Done. If `padding` is not equal to `(kernel_size - 1) // 2`, A `ValueError` will be raised.

File: F-LSeSim/models/kin.py (hunk 2 above)
Comment (st9007a) [reply F-LSeSim/models/kin.py:33]: Done. If `padding` is not equal to `(kernel_size - 1) // 2`, a `ValueError` will be raised.

**Review by Kaminyou — APPROVED** [review]

**Comment by Kaminyou** [comment]
Pass kernel size and kernel type and padding as init arguments

**Pull request merged**